test/terra/backends/aer_simulator/test_wrapper_unitary_simulator.py

```diff
@@ -13,80 +13,100 @@
 UnitarySimulator Integration Tests
 """
 
+from ddt import ddt
 from numpy import exp, pi
 
 from test.terra.reference import ref_1q_clifford
 from test.terra.reference import ref_unitary_gate
 from test.terra.reference import ref_diagonal_gate
 
-from qiskit import execute, assemble, transpile
-from qiskit.providers.aer import UnitarySimulator
+from qiskit import transpile
+from qiskit.providers.aer import UnitarySimulator, AerError
+from test.terra.backends.simulator_test_case import (
+    SimulatorTestCase, supported_devices)
 
 
-class UnitarySimulatorTests:
+@ddt
+class UnitarySimulatorTests(SimulatorTestCase):
```

Comment:
Were these tests even running before?  They don't derive from `unittest.TestCase`, and they were using its methods - I'd have though they weren't being collected, perhaps?  At any rate, certainly better that they're now deriving from a common class.

Reply:
There were used as mixins for a Test case class before, but the mixing code was rather messy so now they are moved to separate test classes.

Reply:
I was supposed to rename this to `TestUnitarySimulator` though

Reply:
Oh, in that case consider this the same comment on the `StatevectorSimulator` test case as well.